src/core/util.h

```diff
@@ -10,13 +10,21 @@
 #define LOVR_VERSION_ALIAS "Maximum Moss"
 
 #ifdef _WIN32
-#define LOVR_NORETURN __declspec(noreturn)
-#define LOVR_THREAD_LOCAL __declspec(thread)
-#define LOVR_RESTRICT __restrict
+  #define LOVR_NORETURN __declspec(noreturn)
+  #define LOVR_THREAD_LOCAL __declspec(thread)
+  #define LOVR_RESTRICT __restrict
+  #if defined(LOVR_BUILDING_SHARED) // building a dll; export symbols
+    #define LOVR_EXPORT __declspec(dllexport)
+  #elif defined(LOVR_BUILDING_EXE) // building exe: keep symbols internal
+    #define LOVR_EXPORT
+  #else // importing this header from a non-lovr project: import the symbols
+    #define LOVR_EXPORT __declspec(dllimport)
+  #endif
```

Comment:
Can you explain the reason why you're varying the __declspec?

Reply:
I added this to our fork a long time ago, trying to remember why it was needed :S it's probably related to us building a lovr.dll and a separate Alloverse.exe that loads it on Windows.  https://stackoverflow.com/a/58031/48125 says it's not required but is more efficient. you'll note enet, glfw, lua etc all do the same thing.

Reply:
I'm not too concerned about the efficiency of LOVR_EXPORT because it is only used by the luaopen_* functions which are called only once at startup.

Although it's not needed right now, it can sometimes be useful to have `__declspec(dllexport)` when building an exe so you can dynamically load symbols from the executable (at one point this was being done to require lovr modules).

If (after potentially resolving the use of LOVR_EXPORT in api.h) you still need to use LOVR_EXPORT, maybe these two BUILDING_* defines could be replaced by just setting LOVR_EXPORT directly from CMakeLists.  The #define here could be guarded by an ifndef.  This would reduce the number of defines to keep track of and maybe simplify the macros here a bit.

I must admit that I'm getting caught up in the "poetry" of having the simpler set of platform-specific defines in this file, and am trying to avoid piling on all the extra windows complexity if possible.

Reply:
I moved it to this file with the "proper" declarations because I saw @mcclure also wanted export macros. I agree it's ugly... I'm not convinced defining it in CMake is less ugly since it's really source code, not a build setting; and it would be needed twice (cmake and tup). Windows just messes with everything, doesn't it :sweat_smile: 

The easiest solution for me is to pull it out once I remove the need for lovrModules, BUT that also feels contraproductive if mcc is gonna need it anyway. eh I'll just revisit once I get to the point where I'm messing with the Windows build...

Reply:
This is gone now since I'm using LOVR_BUILD_WITH_EXPORTED_INTERNALS so I'm resolving this conversation